wiring/inc/spark_wiring_cellular.h

```diff
@@ -74,6 +74,12 @@ class CellularClass : public NetworkClass
 
     CellularSignal RSSI();
 
+    CellularData dataUsage(void);
+
+    CellularData dataUsage(const CellularData &data_set);
```

Comment:
Oh I see it _sets_ the data usage, that wasn't immediately obvious due to my skimming the code and not reading properly. Perhaps a different API name would be helpful? E.g. `Cellular.setDataUsage()`. Thinking forward, I imagine we will need data usage APIs for other networks. How about we rename `CellularData` to be more network-neutral, e.g simply `DataUsage`?

If you could explain the use case for wanting to set the counters to an arbitrary value that might help make it clear to me. 

I'm kind of puzzled why we need a return value at all, and what the counter values of the return value are. The counter values are sent into the modem and the result read out from the modem. I'm puzzled by that! Any reason why the returned counters are not the same as the values passed in? If so then we can simply return the original instance as a reference, since the values are the same.

If the returned counters are truly different, then it would be a good idea to update them in the reference passed in (making the reference non-const). Otherwise the user can easily end up losing that change data by not saving the return value.